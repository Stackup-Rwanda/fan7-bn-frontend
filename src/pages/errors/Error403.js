import React from 'react';
import { Link } from 'react-router-dom';

const Error403 = () => (
  <div className="error">
    <div className="content">
      <h1>403</h1>
      <div className="desc">403 Error Page</div>
      <div className="actions">
        <Link to="/">
          <button type="button">Back to Home</button>
        </Link>
      </div>
    </div>
  </div>
);

export default Error403;
