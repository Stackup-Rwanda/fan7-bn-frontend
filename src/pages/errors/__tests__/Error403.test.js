import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import Error403 from '../Error403';

const history = {
  push: jest.fn(),
  replace: jest.fn(),
  createHref: jest.fn(),
};

describe('Error403 Component Test', () => {
  it('should return true if component exists', () => {
    const wrapper = mount(
      <Router history={history}>
        <Error403 />
      </Router>,
    );

    expect(wrapper.find('.error').exists()).toBe(true);
    expect(wrapper.find('h1').text()).toEqual('403');
  });
});

describe('<Error403 /> shallow rendering tests', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Error403 />);
    expect(tree).toMatchSnapshot();
  });
});
