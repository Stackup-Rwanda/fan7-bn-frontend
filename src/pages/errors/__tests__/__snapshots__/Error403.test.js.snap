// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error403 /> shallow rendering tests matches the snapshot 1`] = `
<div
  className="error"
>
  <div
    className="content"
  >
    <h1>
      403
    </h1>
    <div
      className="desc"
    >
      403 Error Page
    </div>
    <div
      className="actions"
    >
      <Link
        to="/"
      >
        <button
          type="button"
        >
          Back to Home
        </button>
      </Link>
    </div>
  </div>
</div>
`;
